### PR TITLE
[backport/2.11] Debugger readline support

### PR DESCRIPTION
**Pull request by Lord-KA**
The patchset backports test improvements and https://github.com/tarantool/tarantool/commit/855fc6a0cba89d00caeaae6e2e7272c2ee3134e8 (debugger: proper readline support).

### CODE REVIEW

File: changelogs/unreleased/gh-7593-console-debugger.md
Comment (andreyaksenov) [inline changelogs/unreleased/gh-7593-console-debugger.md:3]:
```suggestion
* Introduced `readline` support for the Tarantool debugger (gh-7738).
```

**Review by andreyaksenov — APPROVED** [review]

**Review by igormunkin — CHANGES_REQUESTED** [review]
@Lord-KA, thanks for the patchset! I would like to notice, that the cherry-picked patches must contain the original hash in the commit message (i.e. use `cherry-pick -x` for backporting).

File: changelogs/unreleased/gh-7593-console-debugger.md
Comment (Lord-KA) [reply changelogs/unreleased/gh-7593-console-debugger.md:3]: This is actually a backport, so according to [this](https://github.com/tarantool/tarantool/pull/8880#discussion_r1266507421) discussion, no changes to changelog should be made. But it turns out, the original changelog slipped in unchecked, because I've modified it after it was approved and forgot to rerequest review.

**Comment by Lord-KA** [comment]
> the cherry-picked patches must contain the original hash in the commit message

Done.

File: changelogs/unreleased/gh-7593-console-debugger.md
Comment (Lord-KA) [reply changelogs/unreleased/gh-7593-console-debugger.md:3]: @andreyaksenov, please confirm that it is ok to leave as is.

**Review by andreyaksenov — APPROVED** [review]

**Review by igormunkin — APPROVED** [review]
@Lord-KA, thanks for the fixes! LGTM after all.

**Review by ylobankov — APPROVED** [review]

**Pull request merged**
